lib/inventory_refresh/inventory_collection/builder.rb

```diff
@@ -155,6 +190,12 @@ def to_hash
 
       protected
 
+      def assert_allowed_property(name)
+        unless self.class.allowed_properties.include?(name)
```

Comment:
hm, the allowed_properties should be a set, we don't want O(n) search for every include?